mtags/src/main/scala-2/scala/meta/internal/pc/ScalaPresentationCompiler.scala

```diff
@@ -116,6 +117,111 @@ case class ScalaPresentationCompiler(
 
   def didClose(uri: URI): Unit = {}
 
+  override def semanticTokens(
+      params: VirtualFileParams,
+      capableTypes: util.List[String],
+      capableModifiers: util.List[String]
+   ): CompletableFuture[ju.List[Integer]] = {
+    import scala.collection.mutable.ListBuffer
+    import scala.tools.nsc.ast.parser.Tokens
+    import scala.meta._
+    logger.info("Debug : Scala-PC :")
+
+    val empty: ju.List[Integer] = new ju.ArrayList[Integer]()
+
+    logger.info(" --Before compilerAccess---")
+
+    // val tree = params.text().parse[Source].get
+
+    val buffer = ListBuffer.empty[Integer]
+    var absLine = 0
+    var lastAbsLineOffset = 0
+    var lastAbsLine=0
+    var lastCharStartOffset=0
+
+    val strSep= ",  "
+    val linSep= "\n"
+    val exTokens = params.text().tokenize.get
+    var logString = linSep + params.text()
+
+    logger.info("\n meta.TOKENS : " + exTokens.size.toString)
+    // val testValue= exTokens.map { x => f"${x.structure}%10s -> ${x.getClass}" }.mkString("\n")
+    // logger.info("\n tokenContent : " + testValue + "\n")
+    val compilerTokens = params.text().tokenize.get
```

Comment:
Make sure that you handle a situation if it's empty and the tokenization failed.